### PR TITLE
Mcol 5568 error for invalid conversion

### PR DESCRIPTION
**Pull request by mariadb-SergeyZefirov**
A smallest fix for our inability to detect overflows in the query results.

### CODE REVIEW

**Comment by mariadb-SergeyZefirov** [comment]
Corresponding changes in regression suite: https://github.com/mariadb-corporation/mariadb-columnstore-regression-test/pull/347

File: dbcon/execplan/arithmeticoperator.h [hunk 1]
```diff
@@ -238,6 +239,51 @@ inline void ArithmeticOperator::evaluate(rowgroup::Row& row, bool& isNull, Parse
       break;
 
     case execplan::CalpontSystemCatalog::UBIGINT:
+      {
+        // XXX: this is bandaid solution for specific customer case (MCOL-5568).
+        // Despite that I tried to implement a proper solution: to have operations
+        // performed using int128_t amd then check the result.
+        int128_t x, y;
```
Comment (drrtuy) [inline dbcon/execplan/arithmeticoperator.h:246]: We have TSUInt64 type that I suggest to improve with math result overflow detection so the code can be reused in future if needed.

File: dbcon/execplan/arithmeticoperator.h [hunk 2]
```diff
@@ -238,6 +239,51 @@ inline void ArithmeticOperator::evaluate(rowgroup::Row& row, bool& isNull, Parse
...
+        bool signedLeft = lop->data()->resultType().isSignedInteger();
+        bool signedRight = rop->data()->resultType().isSignedInteger();
+        if (signedLeft)
+        {
+          x = (int128_t)lop->getIntVal(row, isNull);
```
Comment (drrtuy) [inline dbcon/execplan/arithmeticoperator.h]: Did you check if this assignment generates MOVDQA, MOVDQU or MOV?
If it is MOVDQA and getIntValue dst is aligned by 8 bytes there will be SEGV here.

File: dbcon/execplan/arithmeticoperator.h [hunk 3]
```diff
@@ -238,6 +239,51 @@ inline void ArithmeticOperator::evaluate(rowgroup::Row& row, bool& isNull, Parse
...
+        }
+        else
+        {
+          x = (int128_t)lop->getUintVal(row, isNull);
```
Comment (drrtuy) [inline dbcon/execplan/arithmeticoperator.h]:
```suggestion
          x = static_cast<int128_t(lop->getUintVal(row, isNull));
```

File: mysql-test/columnstore/basic/r/MCOL-5568-out-of-range.result [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+CREATE TABLE test_mult (
+indemnity_paid INT(11),
+n_clms TINYINT(3) UNSIGNED
+) ENGINE=COLUMNSTORE;
+INSERT INTO test_mult (indemnity_paid, n_clms) VALUES (-10, 1);
+SELECT indemnity_paid, n_clms, indemnity_paid * n_clms FROM test_mult;
+ERROR HY000: Internal error: MCS-2053: The result is out of range for function "*" using value(s): -10.000000 1.000000
+SELECT indemnity_paid, n_clms, indemnity_paid + n_clms FROM test_mult;
+ERROR HY000: Internal error: MCS-2053: The result is out of range for function "+" using value(s): -10.000000 1.000000
+SELECT indemnity_paid, n_clms, (indemnity_paid + 9) + n_clms FROM test_mult;
```
Comment (drrtuy) [inline mysql-test/columnstore/basic/r/MCOL-5568-out-of-range.result:13]: I would appreciate if you extend the code and the examples for combinations of: smaller unsigned SQL datatypes, e.g. UBINT, USMALLINT, UTINYINT, *,/ operations and a signed type as the second argument.

**Review by drrtuy — CHANGES_REQUESTED** [review]
Plz address the comments and the one I shared in Slack.

**Review by drrtuy — APPROVED** [review]
utACK

**Comment by mariadb-LeonidFedorov** [comment]
PR should be duplicated for branch `stable-23.10`

**Pull request merged**